### PR TITLE
Tom/react router

### PR DESCRIPTION
**Pull request by tfettig22**
What's this PR do?
- Convert Modal component to class-based.  Now when the  Modal mounts, it does a fetch for the single movie details, which can then be used to populate the Modal details.  URL properly updates using react router.  Needs some css work now to make things format correctly
- Added a conditional to the Modal render so that if the fetch is not completed before the render happens, it renders a 'Loading..." message, and then when the fetch updates the state changes and the Modal re-renders with the movie details.

- [x] New feature (adds functionality)
- [x] I tested my changes in the browser
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new bugs

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-modal": "^3.15.1",
+    "react-router-dom": "^5.3.0",
```
Comment (GOECHA) [inline package.json:13]: initialize router- viewed
I

File: src/components/App/App.js [hunk 2]
```diff
@@ -5,57 +5,33 @@ import Header from '../Header/Header'
...
     super();
     this.state ={
         movies: [],
-        result: ``,
-        showModal: []
       }
-    };
+  };
 
-    componentDidMount = () => {
```
Comment (GOECHA) [inline src/components/App/App.js:21]: > 🟢 Nice refactor!

File: src/components/App/App.js [hunk 3]
```diff
@@ -5,57 +5,33 @@ import Header from '../Header/Header'
...
-                ) : (
-                  <>
-                   <img className="background" src='https://image.tmdb.org/t/p/original//pq0JSpwyT2URytdFG0euztQPAyR.jpg' alt=''/>
-                    <MovieContainer movies={this.state.movies} selectAMovie={this.selectAMovie}/> 
-                   
-                   </>
-                )}  
-          
-        </main>
+      <main className = 'App'>
```
Comment (GOECHA) [inline src/components/App/App.js:28]: > 🟢 💚 🟢 Nice refactor!

File: src/components/Card/Card.js [hunk 4]
```diff
@@ -3,7 +3,7 @@ import './Card.css';
 
 const Card = ({ logo, posterImage, rating, id, selectAMovie }) => {
   return (
-    <div className='movie-card' onClick={(event) => selectAMovie(event)}>
```
Comment (GOECHA) [inline src/components/Card/Card.js:6]: > 🟢 Nice refactor!

File: src/components/Modal/Modal.js [hunk 5]
```diff
@@ -1,53 +1,56 @@
-import React from 'react';
+import React, { Component } from 'react';
 import './Modal.css';
-import logo from '../../images/rancid-tomatillos.png'
 import dayjs from 'dayjs'
+import { NavLink } from 'react-router-dom'
+import { getAllData } from '../../api-calls';
 
-const Modal = ({ props, backToHome }) => {
```
Comment (GOECHA) [inline src/components/Modal/Modal.js:6]: This refactor looks great.

File: src/components/Modal/Modal.js [hunk 6]
```diff
@@ -1,53 +1,56 @@
...
-    overview,
-    genres,
-    budget,
-    poster_path,
-    runtime,
-    revenue,
-    tagline,
-  } = movie;
-  
+class SingleMovie extends Component {
```
Comment (GOECHA) [inline src/components/Modal/Modal.js:7]: Great job with functionality, this looks clean and is very readable. I like how we changed this functional component into a class component. If there is time, I would like to consider a refactor with hooks.

File: src/components/MovieContainer/MovieContainer.js [hunk 7]
```diff
@@ -2,24 +2,26 @@ import React from 'react'
...
     return (
-      <Card
-        logo={logo}
-        posterImage={movie.poster_path}
-        rating={movie.average_rating.toFixed(1)}
-        id={movie.id}
-        key={movie.id}
-        selectAMovie={selectAMovie}
-      />
+      <NavLink to={`/modal/${movie.id}`} key={movie.id}>
```
Comment (GOECHA) [inline src/components/MovieContainer/MovieContainer.js:16]: - reviewed move NavLink to Card to direct to Modal

File: src/index.js [hunk 8]
```diff
@@ -2,12 +2,12 @@ import React from 'react';
 import ReactDOM from 'react-dom/client';
 import './index.css';
 import App from './components/App/App';
-
+import { BrowserRouter } from 'react-router-dom';
```
Comment (GOECHA) [inline src/index.js:5]: - Thank you for adding BrowserRouter and getting us set up for router!

**Review by GOECHA — APPROVED** [review]
Great job with functionality, this looks clean and is very readable.
> 🟢 Nice refactor!

File: src/components/MovieContainer/MovieContainer.js (hunk 7 above)
Comment (Maia-Cochran) [reply src/components/MovieContainer/MovieContainer.js:16]: I love this piece! Makes a lot of sense for where and when the NavLink should be present in the code :-)

File: src/index.js (hunk 8 above)
Comment (Maia-Cochran) [reply src/index.js:5]: To echo Chantal... thank you !!

**Pull request merged**